crates/loader/src/local/config.rs

```diff
@@ -22,12 +22,43 @@ pub(crate) type RawAppManifestAnyVersionPartial = RawAppManifestAnyVersionImpl<t
 pub(crate) type RawComponentManifestPartial = RawComponentManifestImpl<toml::Value>;
 
 /// Container for any version of the manifest.
-#[derive(Clone, Debug, Deserialize, Serialize)]
-#[serde(tag = "spin_version")]
+#[derive(Clone, Debug, Deserialize)]
+#[serde(untagged)]
 pub enum RawAppManifestAnyVersionImpl<C> {
-    /// A manifest with API version 1.
-    #[serde(rename = "1")]
-    V1(RawAppManifestImpl<C>),
+    /// Old Spin manifest versioned with `spin_version` key
+    V1Old {
+        /// Version key name
+        spin_version: FixedStringVersion<1>,
+        /// Manifest
+        #[serde(flatten)]
+        manifest: RawAppManifestImpl<C>,
+    },
+    /// New Spin manifest versioned with `spin_manifest_version` key
+    V1New {
+        /// Version key name
+        spin_manifest_version: FixedStringVersion<1>,
+        /// Manifest
+        #[serde(flatten)]
+        manifest: RawAppManifestImpl<C>,
+    },
+}
+
+impl<C> RawAppManifestAnyVersionImpl<C> {
+    /// Get V1 manifest
+    pub fn into_v1(self) -> RawAppManifestImpl<C> {
+        match self {
```

Comment:
This is great - the convenience of an `into_v1` with the safety of an exhaustive match.  Thanks!